### ISSUE TITLE
Maintain org indentation

### ISSUE THREAD

**Issue by mjhoy** [rksm/org-ai#18]
AI chat within `begin_ai` blocks doesn't seem to maintain `org-indent-mode` in a way that makes my OCD go crazy 😂. Looking at the code, I don't totally understand why this is... it seems to just use `insert` and when I evaluate `(insert "\n\n")` locally, for instance, I can't reproduce the behavior.

**Comment by rksm** [comment]
Yeah seeing this as well. When I disable/enable org-indent-mode it seems to get fixed. It might be related to the org-element-cache issue? https://github.com/rksm/org-ai/issues/16. Hmm.

**Comment by shamiv** [comment]
I am definitely not the org-mode expert/maintainer you're looking for, but running this inside the block after the output has been inserted fixes it for me. I was looking for a way to have it run after the text insertion is complete, but I didn't manage yet (there is no hook for that?)

```elisp
  (defun org-ai-dirty-indent-fix ()
    (interactive)
    (save-excursion
    (evil-org->
     (re-search-backward "^\\#\\+begin_ai" nil t)
     (re-search-forward "^\\#\\+end_ai" nil t) 2)
    (evil-undo 1)))
```

**Comment by LemonBreezes** [comment]
I personally just use `C-x x g`, `revert-buffer-quick` if I notice the indentation is off.

**Comment by shamiv** [comment]
Is there a way run revert-buffer-quick after org-ai is done? Some hook that runs functions after the response has been inserted?

**Comment by rksm** [comment]
You can do something like

```emacs-lisp
(add-hook 'org-ai-after-chat-insertion-hook
          (lambda (type text) ; type will be 'text, 'role or 'end
            (when (eq type 'end)
              (save-buffer)
              (revert-buffer-quick))))
```

**Comment by tavisrudd** [comment]
I've been using the following hook to solve this issue:
```elisp
(defun dss/-org-ai-after-chat-insertion-hook (type _text)
  (when (and (eq major-mode 'org-mode) (eq type 'end))
    (org-indent-indent-buffer)))
(add-hook 'org-ai-after-chat-insertion-hook #'dss/-org-ai-after-chat-insertion-hook)
```
